sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -150,8 +161,24 @@ class RocksDB(
           metadata.numKeys
         }
         numKeysOnWritingVersion = numKeys
-        numKeysOnLoadedVersion = numKeys
 
+        for (v <- latestSnapshotVersion + 1 to version) {
```

Comment:
Maybe add a small comment for what this snippet is doing ?

Reply:
Sure.